### ISSUE TITLE
sys.stdin, sys.stdout, sys.stderr are of type io.TextIOWrapper

### ISSUE THREAD

**Issue by bdrung** [python/typeshed#10093]
Example code:
```python
#!/usr/bin/python3
import sys
print(type(sys.stdin))
print(type(sys.stdout))
print(sys.stdin.detach())
```
This results in following output:
```
$ ./stdin.py 
<class '_io.TextIOWrapper'>
<class '_io.TextIOWrapper'>
<_io.BufferedReader name='<stdin>'>
```
But mypy complains:
```
$ mypy stdin.py 
stdin.py:5: error: "TextIO" has no attribute "detach"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by bdrung** [comment]
Apport uses `sys.stdout.detach()` and it also redefines it with `sys.stdin = os.fdopen(int(fds[0]), "r")` at some point.

**Comment by Akuli** [comment]
While it is common to redefine `sys.stdout = io.StringIO()` or `sys.stderr = io.StringIO()`, maybe doing that isn't so common for stdin and we could have a more specific `sys.stdin` type.

**Comment by TeamSpen210** [comment]
Since it is expected that libraries could redefine them, perhaps this error is good since it'll force you to use `isinstance()` or `hasattr()`, and handle the case that `detach()` isn't present?

**Comment by AlexWaygood** [comment]
We previously discussed this in https://github.com/python/typeshed/pull/5623.

I guess I'd be open to having a more specific type for `sys.stdin`, as @Akuli suggests. But it does seem a bit weird to have a different type for `sys.stdin` than the one we use for the other two. That feels like it might cause yet more user confusion.

Whatever we do here, we should probably add a comment to the stub, at the very least. It's hard to properly document the decisions we make at typeshed, but we can do better than what we have at the moment:

https://github.com/python/typeshed/blob/6f6b1c9135edbe7e6b16b06070f3b12c7bee3a5a/stdlib/sys.pyi#L60-L62
